extension/src/utils/security.ts

```diff
@@ -1,9 +1,25 @@
 import { randomBytes, X509Certificate } from 'crypto';
 import forge from 'node-forge';
 
-export function createSelfSignedCert(commonName: string = 'localhost') {
+interface SelfSignedCert {
+    key: string;
+    cert: string;
+    certBase64: string;
+}
+
+export async function createSelfSignedCertAsync(commonName: string = 'localhost'): Promise<SelfSignedCert> {
```

Comment:
Dumb question, but this was async before and now it's async. Why did that change?

Reply:
Not a dumb question. It takes much longer than 2x the time to generate a 4096 bit key pair compared to 2048, so it seemed worth it to make this step asynchronous at this point. [node-forge](https://www.npmjs.com/package/node-forge#rsa) also recommends async generation.